src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -58,6 +58,8 @@ namespace winrt
 
 namespace winrt::TerminalApp::implementation
 {
+    bool TerminalPage::_hasShownTerminalTitleInTitlebar{ false };
```

Comment:
ah - having a global for this may make it work poorly when there are multiple windows open. All windows are served by the same instance of WindowsTerminal.exe 🙂

Reply:
good point, didn't consider this; Let me restructure, maybe i can put this in terminal window somehow

Reply:
Hi, I have come up with a more concise/cleaner approach but its potentially slower, however, it does work with 4+ windows, could you provide guidance if this approach is against best practice (I had to introduce a getTitle to the basewindow.h) @DHowett